packages/cozy-client/src/authentication/mobile.js

```diff
@@ -1,48 +1,83 @@
-        options
-      )
+const authenticateWithSafari = url => {
+  return new Promise((resolve, reject) => {
+    window.SafariViewController.show(
+      {
+        url: url,
+        transition: 'curl' // (this only works in iOS 9.1/9.2 and lower) unless animated is false you can choose from: curl, flip, fade, slide (default)
+        // enterReaderModeIfAvailable: readerMode, // default false
+        // tintColor: "#00ffff", // default is ios blue
+        // barColor: "#0000ff", // on iOS 10+ you can change the background color as well
+        // controlTintColor: "#ffffff" // on iOS 10+ you can override the default tintColor
+      },
+      // this success handler will be invoked for the lifecycle events 'opened', 'loaded' and 'closed'
+      result => {
+        if (result.event === 'closed') {
+          reject(new Error(REGISTRATION_ABORT))
+        }
+      },
+      error => {
+        console.log('KO: ' + error)
+        reject(new Error(REGISTRATION_ABORT))
+      }
+    )
 
-      const removeListener = () => {
-        inAppBrowser.removeEventListener('loadstart', onLoadStart)
-        inAppBrowser.removeEventListener('exit', onExit)
+    const handle = window.handleOpenURL
+    window.handleOpenURL = url => {
+      window.SafariViewController.hide()
+      resolve(url)
+      if (handle) {
+        window.handleOpenURL = handle
       }
+    }
+  })
+}
 
-      const onLoadStart = ({ url }) => {
-        const accessCode = /\?access_code=(.+)$/.test(url)
-        const state = /\?state=(.+)$/.test(url)
+const authenticateWithInAppBrowser = url => {
+  return new Promise((resolve, reject) => {
+    const target = '_blank'
+    const options = 'clearcache=yes,zoom=no'
+    const inAppBrowser = window.cordova.InAppBrowser.open(url, target, options)
 
-        if (accessCode || state) {
-          resolve(url)
-          removeListener()
-          inAppBrowser.close()
-        }
-      }
+    const removeListener = () => {
+      inAppBrowser.removeEventListener('loadstart', onLoadStart)
+      inAppBrowser.removeEventListener('exit', onExit)
+    }
 
-      const onExit = () => {
-        reject(new Error(REGISTRATION_ABORT))
+    const onLoadStart = ({ url }) => {
+      const accessCode = /\?access_code=(.+)$/.test(url)
+      const state = /\?state=(.+)$/.test(url)
+
+      if (accessCode || state) {
+        resolve(url)
         removeListener()
         inAppBrowser.close()
       }
+    }
 
-      inAppBrowser.addEventListener('loadstart', onLoadStart)
-      inAppBrowser.addEventListener('exit', onExit)
-    })
+    const onExit = () => {
+      reject(new Error(REGISTRATION_ABORT))
+      removeListener()
+      inAppBrowser.close()
+    }
+
+    inAppBrowser.addEventListener('loadstart', onLoadStart)
+    inAppBrowser.addEventListener('exit', onExit)
+  })
+}
+
+export const authenticateWithCordova = async url => {
+  if (isIOSApp() && (await hasSafariPlugin())) {
+    return authenticateWithSafari(url)
```

Comment:
Why don't we use this plugin also on Android?

Reply:
I copied the code from the old cozy-client https://github.com/cozy/cozy-drive/blob/master/src/lib/cozy-client/authentication/mobile.js#L31 ;)

You think we can activate it ?

Reply:
After discussing on mattermost we decided to activate it on Android